README.md

```diff
@@ -45,3 +76,44 @@ include:
 *   Thomas Honeyman (@thomashoneyman) - Infrastructure
 *   Justin Woo (@justinwoo) - Package Sets
 *   Fabrizio Ferrai (@f-f) - Package Sets
+
+Collaborators are encouraged to help out with maintenance in any of the
+following ways (or however else they see fit):
+
+*   Merging PRs if they are ready to be merged (see below)
+*   Managing the issue tracker (labelling, closing, or otherwise tidying up
+    issues)
+*   Discussing proposed features or bug reports
+*   Sending PRs
+*   Reviewing PRs
+
+## Making Decisions
+
+Most decisions which directly concern the language, compiler, or core libraries
+should receive the approval of the core owners. For example, this would apply
+to:
+
+*   Feature requests
+*   Concerns raised during pull request review
+*   Adding new collaborators
+*   Promoting collaborators to owners
+
+For a decision to be made, it should receive the approval of all core owners
+who choose to participate in the discussion. In essence, every core owner has a
+veto (which they are expected to use responsibly).
+
+Not all core owners are required to participate in every discussion -- it is
+expected that most discussions will only involve a subset of the core owners.
+Before any particular decision is finalized, sufficient time should be given so
+that people have an opportunity to object, especially for decisions which would
+be difficult to reverse later.
```

Comment:
This sounds good to me, but I'd like to specify a timeout for engaging in the discussion.
The reason for this wish is that otherwise it would be ambiguous when a decision is actually effective if not everyone is involved in the discussion, and the risk is for things to get stuck in the "approval limbo"
It doesn't matter what the timeout is exactly (1 week? 2 weeks? 1 month?) because it can be changed if it doesn't fit - it just matters that it's there and it's not unreasonably long (e.g. 1 year would not work)

Two more small notes:
1. we might want to have something like a set of labels that we attach to PRs and issues to signal the following things:
    - "needs approval from the core team"
    - "voting started"
    - "approved"
    - "rejected"
  
    If this sounds sensible, then we should specify them here so that people involved in such discussions can always refer to this document for reference on how things work.
2. there is a meta aspect involved in all of this, e.g. what's the time frame for this PR to be considered "approved" if not all the owners participate in the discussion? (this is a rhetoric question - the answer is "we approve this one with the implicit rules we had so far")
  However, once this PR is merged then we'll have a documented way to handle approvals/merging/etc. I feel there is a small ambiguity about "is this document included in the rules?" - intuitively it is, but I'd like to see it specified in the "list of decisions that need approval" above

Reply:
I would like to clarify the workflow for issues and PRs, although I think labels might be redundant. Just for the sake of discussion, let's assume that all issues are either feature requests or bug reports, and that people agree on how to categorise them. Then there are three different types of tickets: bug reports, feature requests, and PRs. For bug reports, I don't think we need consider them to be something which need to be voted on, they just need to be validated (or otherwise closed). For feature requests, we've had a vague policy for a while (albeit not always applied consistently and not written down anywhere) that feature requests should go to Discourse first, so that the issue tracker is kept for things that a) represent dev work on the compiler, and b) we are reasonably sure we want to work on. We now have https://github.com/purescript/purescript/issues/3832 for addressing this. For PRs, I think the process is covered already by code review approvals?

Reply:
Including changes to this document as things which the "making decisions" process applies to sounds good to me.

Reply:
How about the timeout for owners to chime in on things? (which was my main point, sorry if it got lost in my minor comments)

Reply:
For PRs, I think we generally wait a few days max after an approval for another owner to voice feedback. I would say 3 days.

Reply:
Yeah, this is more difficult to answer. In general the approach I've taken is to use my judgement - the length of time I'd wait before proceeding with something depends on the potential for decisions to have larger, longer-term consequences. If it's a really big decision I might give it months. I would like people to have time to properly consider proposals and write up any objections properly without feeling rushed, because in my experience it can take a little while to do that.

Reply:
Is "approval limbo" a big enough problem that we need to explicitly address it here? I'm not sure it is, but I'm open to being convinced.

Reply:
On the other hand I'd say 3 days is very small for a change that might heavily affect the language/documentation (e.g. what if someone is on vacation for weeks?).
It might be worthwile having different tiers, e.g. 3 days for lightweight decisions and some weeks for big ones, so that people have time to respond

Reply:
```suggestion
be difficult to reverse later. The current minimum time to wait is three days.
```